functional_widget/lib/function_to_widget_class.dart

```diff
@@ -51,31 +52,31 @@ class FunctionalWidgetGenerator
         .toString();
   }
 
-  FunctionElement _checkValidElement(Element element) {
-    if (element is! FunctionElement) {
-      throw InvalidGenerationSourceError(
-        'Error, the decorated element is not a function',
-        element: element,
-      );
-    }
-    var function = element as FunctionElement;
-    if (function.isAsynchronous ||
-        function.isExternal ||
-        function.isGenerator ||
-        function.returnType?.displayName != 'Widget') {
-      throw InvalidGenerationSourceError(
-        'Invalid prototype. The function must be synchronous, top level, and return a Widget',
-        element: function,
-      );
-    }
-    final className = _toTitle(function.name);
-    if (className == function.name) {
-      throw InvalidGenerationSourceError(
-        'The function name must start with a lowercase',
-        element: function,
-      );
+  FunctionElement _checkValidElement(Element function) {
+    if (function is FunctionElement) {
```

Comment:
Why the refactor of this method?

Reply:
that we we can remove this line 
`var function = element as FunctionElement;`
and the more important part of the function is at the top (the type checking in the if will do the job of `as FunctionElement`)
*I half meant to remove it, I sometimes refactor little things so that I understand them more easily but I can remove it if you prefer*

Reply:
I see! 😄 
I prefer the older version though, as it is easier to read (linear, no indentation, fail fast).

The cast is more of an issue with Dart type inference than with the code itself.

Reply:
ok cool, I'll put it back :-)

Reply:
yeah, the indentation is nicer, Dart type inference wouldn't fix it though as it is only checking what it is not and not what it is (if that makes sense!)

Reply:
! FunctionElement could be any element
= FunctionElement can only be FunctionElement

Reply:
its a minor one though...kind of thing I spend too much of my time thinking about!

Reply:
what I said makes no sense!  if(element is! FunctionElement){} the rest means it must be a FunctionElement...yeah, hopefully the type inference will get better ;-)